demo/react-native/src/App.tsx

```diff
@@ -0,0 +1,375 @@
+//
+// Copyright 2022 Picovoice Inc.
+//
+// You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+// file accompanying this source.
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//
+
+import React, {Component} from 'react';
+import {
+  EventSubscription,
+  NativeEventEmitter,
+  PermissionsAndroid,
+  Platform,
+  SafeAreaView,
+  ScrollView,
+  StatusBar,
+  TouchableOpacity,
+} from 'react-native';
+import {StyleSheet, Text, View} from 'react-native';
+
+import {Leopard, LeopardErrors} from '@picovoice/leopard-react-native';
+import {
+  VoiceProcessor,
+  BufferEmitter,
+} from '@picovoice/react-native-voice-processor';
+
+import Recorder from './Recorder';
+
+enum UIState {
+  loading,
+  init,
+  recording,
+  processing,
+  transcribed,
+  error,
+}
+
+type Props = {};
+type State = {
+  appState: UIState;
+  errorMessage: string | null;
+  transcription: string;
+  recordSeconds: number;
+  processSeconds: number;
+};
+
+export default class App extends Component<Props, State> {
+  _leopard?: Leopard;
+  _accessKey: string = '${YOUR_ACCESS_KEY_HERE}'; // AccessKey obtained from Picovoice Console (https://picovoice.ai/console/)
+
+  _recorder: Recorder = new Recorder();
+  _voiceProcessor?: VoiceProcessor;
+  _bufferListener?: EventSubscription;
+  _bufferEmitter?: NativeEventEmitter;
+
+  _recordInterval?: NodeJS.Timer;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      appState: UIState.loading,
+      errorMessage: null,
+      transcription: '',
+      recordSeconds: 0.0,
+      processSeconds: 0.0,
+    };
+  }
+
+  componentDidMount() {
+    this.init();
+  }
+
+  componentWillUnmount() {
+    if (this.state.appState === UIState.recording) {
+      this._stopProcessing();
+    }
+    if (this._leopard !== undefined) {
+      this._leopard.delete();
+      this._leopard = undefined;
+    }
+  }
+
+  async init() {
+    try {
+      this._leopard = await Leopard.create(
+        this._accessKey,
+        'leopard_params.pv',
+      );
+      this._voiceProcessor = VoiceProcessor.getVoiceProcessor(
+        512,
+        this._leopard.sampleRate,
+      );
+      this._bufferEmitter = new NativeEventEmitter(BufferEmitter);
+      this._bufferListener = this._bufferEmitter.addListener(
+        BufferEmitter.BUFFER_EMITTER_KEY,
+        async (buffer: number[]) => {
+          await this._recorder.writeSamples(buffer);
```

Comment:
How are we managing the state of this file? You may want to be more clear about what state the file is in rather than relying on writing the header to restart the file. It may also be good to add an error state, so that you don't write to the file if there's been a problem.